### PR TITLE
Vector2D corrections

### PR DESCRIPTION
**Pull request by Quazistax**
- cross product corrected
- check for computation floating point error in angleBetween (cos can
  turn out to be +-1.0000000000000002)

### CODE REVIEW

File: lib/Vector2D.js [hunk 1]
```diff
@@ -151,6 +151,8 @@ Vector2D.prototype.divide = function(scalar) {
  */
 Vector2D.prototype.angleBetween = function(that) {
     var cos = this.dot(that) / (this.length() * that.length());
+    if (cos < -1) cos = -1;
+    else if (cos > 1) cos = 1;
```
Comment (thelonious) [inline lib/Vector2D.js]: Didn't know that.

BTW, I prefer if/else statements to always use curly braces. Can you change this to following?:

```
if (cos < -1) {
  cos = -1;
}
else if (cos > 1) {
  cos = 1;
}
```

I guess if there are going to be contributors, I should put up a style guide.

**Comment by thelonious** [comment]
Thanks for catching the cross product issue. I thought I had corrected that in all places, but apparently not.

I'll merge and update related modules once the formatting has been updated.

File: lib/Vector2D.js (hunk 1 above)
Comment (thelonious) [inline lib/Vector2D.js]: alternately, I often use `cos = Math.max(-1, Math.min(1, cos));` but that does add some function call overhead

**Comment by thelonious** [comment]
Thanks again.

**Comment by Quazistax** [comment]
No problem. Thank you for sharing your code ;)

**Pull request merged**